src/cls/rbd/cls_rbd.cc

```diff
@@ -2675,6 +2693,51 @@ int metadata_get(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
   return 0;
 }
 
+int get_snapshot_limit(cls_method_context_t hctx, bufferlist *in,
+		       bufferlist *out)
+{
+  int rc;
+  uint64_t snap_limit;
+
+  rc = read_key(hctx, "snap_limit", &snap_limit);
+  if (rc == -ENOENT) {
+    rc = 0;
+    ::encode(UINT64_MAX, *out);
+  } else {
+    ::encode(snap_limit, *out);
+  }
+
+  CLS_LOG(20, "read snapshot limit %lu", snap_limit);
+  return rc;
+}
+
+int set_snapshot_limit(cls_method_context_t hctx, bufferlist *in,
+		       bufferlist *out)
+{
+  int rc;
+  int new_limit;
```

Comment:
Yeah, I don't like that asymmetry either. I wanted the user to be able to specify a negative value on the command line instead of remembering UINT64_MAX and I didn't want the CLI to contain the logic to translate it. I can put it there if that's ok, though.